templates/extract_bin_file.ipynb

```diff
@@ -1,43 +1,101 @@
-    "from mibi_bin_tools import bin_files"
+    "from mibi_bin_tools import bin_files, io_utils"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e2ba233e-6904-409c-b324-162dc9ff10e7",
+   "metadata": {},
+   "source": [
+    "## Required variables\n",
+    "\n",
+    " - `run_name` should contain the exact name of the MIBI run to extract from\n",
+    " - `panel_path` should point to a panel csv specifying the targets on your panel. For an example, see the [example panel file](https://github.com/angelolab/toffy/blob/main/files/example_panel_file.csv)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 4,
+   "execution_count": 10,
+   "id": "2726b808-9378-41e5-a8d1-788a693e6937",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# data_dir should contain *.bin's and *.json's, no subdirs for now\n",
-    "base_dir = 'D:\\\\Data\\\\2021-12-12_run_name'\n",
+    "# the name of the run\n",
+    "run_name = 'YYYY-MM-DD_run_name'\n",
+    "\n",
+    "# path to user panel\n",
+    "panel_path = 'C:\\\\Users\\\\Customer.ION\\\\Documents\\\\panel_files\\\\my_cool_panel.csv'\n",
+    "panel = pd.read_csv(panel_path) \n",
+    "\n",
+    "# path to the directory containing the FOV bin files\n",
+    "base_dir = os.path.join('D:\\\\Data', run_name) \n",
     "\n",
     "# create a new folder to hold extracted files\n",
-    "img_out_dir = os.path.join(base_dir, 'extracted')\n",
-    "if not os.path.exists(img_out_dir):\n",
-    "    os.makedirs(img_out_dir)\n",
+    "extraction_dir = os.path.join('D:\\\\Extracted_Images', run_name)\n",
+    "if not os.path.exists(extraction_dir):\n",
+    "    os.makedirs(extraction_dir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8767f85a-6c91-4e4e-9fe1-2a1603fdf384",
+   "metadata": {},
+   "source": [
+    "## Extract FOV bin files\n",
     "\n",
-    "# Column names should be uppercased: 'Mass', 'Target', 'Start', 'Stop'\n",
-    "panel = pd.read_csv(os.path.join(base_dir, 'panel_full.csv'))\n",
+    "The default for `extract_bin_files` is to not extract the intensity data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "cd6eda88-3003-496f-8c66-1697d570f05d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      " \n"
+     ]
+    }
+   ],
+   "source": [
+    "    # retrieve all fov names from base_dir and extracted fovs from extraction_dir\n",
+    "fovs = io_utils.remove_file_extensions(io_utils.list_files(base_dir, substrs='.bin'))\n",
+    "extracted_fovs = io_utils.list_folders(extraction_dir, substrs='fov')\n",
     "\n",
-    "bin_files.extract_bin_files(base_dir, img_out_dir, include_fovs=None, panel=panel, write_parallel=False)"
+    "# extract missing fovs to extraction_dir\n",
+    "missing_fovs = list(set(fovs).difference(extracted_fovs))\n",
+    "bin_files.extract_bin_files(base_dir, extraction_dir, include_fovs=missing_fovs, panel=panel, intensities=False)"
```

Comment:
We should have a variable called `extract_intensities` that is hard-coded to `False`, but the user can overwrite to `True`. This line would then become:

```
bin_files.extract_bin_files(base_dir, extraction_dir, include_fovs=missing_fovs, panel=panel, intensities=extract_intensities)
```

Notebook documentation should also describe what `extract_intensities` does.